lib/src/__tests__/DatePicker/Calendar.test.tsx

```diff
@@ -24,6 +23,40 @@ describe('Calendar', () => {
     component.find('CalendarHeader').prop<any>('onMonthChange')(utilsToUse.date());
     expect(onMonthChangeMock).toHaveBeenCalled();
   });
+
+  it('Should not display loading indicator when synchronous', () => {
```

Comment:
The tests must be moved to e2e. Or at least using mount.

Reply:
Thanks for the feeback! 
I could definitely move the tests but I'm wondering if they really are e2e tests? 

Except from the confusing test names (who clearly mentions visibility of elements, which is kindof misleading and should be changed), the tests really only checks so that the state of the component is changed properly.

Mount is probably more suitable for these kind of tests, I agree, but does state manipulation in components belong to e2e tests?

Reply:
Yeep, it is more integration then e2e. 
We are on the road to testing public interface rather then internal component logic

Reply:
mount tests just named `e2e` for some reason

Reply:
Moved the tests and modified them now